storage/store_test.go

```diff
@@ -1106,3 +1107,49 @@ func TestRaftNodeID(t *testing.T) {
 		}()
 	}
 }
+
+// lookupAndAddRandomRanges adds numRanges ranges with randomly
+// generated start/end keys. Before adding a range, looks up if there is
+// any range that has the same key.
+func lookupAndAddRandomRanges(numRanges int, b *testing.B) {
+	store, _, stopper := createTestStore(b)
+	defer stopper.Stop()
+
+	// Remove range 1.
+	rng1, err := store.GetRange(1)
+	if err != nil {
+		b.Fatal(err)
+	}
+	if err := store.RemoveRange(rng1); err != nil {
+		b.Fatal(err)
+	}
+
+	for i := 0; i < numRanges; i++ {
```

Comment:
You should probably do a `b.ResetTimer()` just before the for loop so that the initialization work doesn't count towards  the time.